src/main/java/uk/ac/cam/cl/dtg/segue/dao/users/PgUsers.java

```diff
@@ -762,7 +762,16 @@ private RegisteredUser buildRegisteredUser(final ResultSet results) throws SQLEx
         u.setGivenName(results.getString("given_name"));
         u.setEmail(results.getString("email"));
         u.setRole(results.getString("role") != null ? Role.valueOf(results.getString("role")) : null);
-        u.setDateOfBirth(results.getDate("date_of_birth"));
+        java.sql.Date dateOfBirth = results.getDate("date_of_birth");
+        // So, DOB is a date in the database.
+        // Because Java is Java, and this code goes back to 1996, the date is interpreted as a datetime at midnight
+        // in the local timezone. This is obviously insane, but there you go.
+        // It works on the servers because they are sensibly set to a UTC timezone. However, it is broken on local
+        // machines where the timezone is London time.
+        // The front-end expects a timestamp in UTC, so this line uses four deprecated functions to make one. The joy!
+        Date utcDate = new Date(
+            Date.UTC(dateOfBirth.getYear(), dateOfBirth.getMonth(), dateOfBirth.getDate(), 0, 0, 0));
+        u.setDateOfBirth(utcDate);
```

Comment:
Got to love the old programming problems - thinking back to the millennium bug. 

We have the joda library in our dependency list already so I think that might have nicer less deprecated methods. I can see this fixes the bug so the use of joda is only a suggestion.

Reply:
Java has since also added LocalDate which is worth having a look at. I think its designers were heavily influenced by the joda library.
https://docs.oracle.com/javase/8/docs/api/java/time/LocalDate.html

Reply:
Just to note, I didn't see any particularly neat way to do this, and you have to end up with a Date object in the right timezone so I thought just keeping it simple was safer.